src/media/mod.rs

```diff
@@ -674,7 +683,7 @@ impl<'a> Future for UploadFuture<'a> {
     }
 }
 
-/// An error wrapper for `UploadFuture`, noting what stage of the upload an error occurred at.
+/// An wrapper for `UploadFuture` error, noting at which stage of the upload the error occurred at.
```

Comment:
would prefer "for `UploadFuture` errors", plural

Reply:
ok

Reply:
Btw wouldn't it also make sense to Deref to underlying error?

Reply:
Huh, i like that idea, but since the underlying `Error` is just an enum with no real methods of its own, it wouldn't really make sense from a design perspective. If you want to match on the enum, you're going to be matching on `err.error` regardless; the only thing `Deref` would add in that case is the ability to use `&*` instead.